mars-rover-new/src/mars_rover_new/core.clj

```diff
@@ -111,8 +155,25 @@
             (println "Command sequence error, please try again.")))
         (process-user-input)))))
 
+;; process-cmds uses fn-map to get the core functions corresponding to L, R and M commands
+(defn process-cmds
+  "Reduces over a list of functions and builds up the rover's final position"
+  [mars-rover-data]
+  (reduce (fn [new-mars-rover-data cmd] ((get fn-map cmd) new-mars-rover-data))
+          mars-rover-data
+          (map str (get mars-rover-data :cmd-sequence))))
+
+(defn display-result
+  "Display final rover position and check if position is within plateau area"
+  [mars-rover-data]
+  (let [{x :x y :y d :direction} (get mars-rover-data :rover-position)
+        {max-x :x max-y :y} (get mars-rover-data :plateau-size)]
+    (if (and (in-range? x 0 max-x) (in-range? y 0 max-y))
```

Comment:
Consider moving this logic outside somewhere else.
Generally `display-result` should be a "dumb" function which assumes that whatever we pass to it is correct.

This loops into the Single Responsibility Principle.